src/app/add-clients/helperFunctions.tsx

```diff
@@ -0,0 +1,254 @@
+import React from "react";
+import { Database } from "@/database_types_file";
+import { booleanGroup } from "@/components/DataInput/inputHandlerFactories";
+import { SelectChangeEvent } from "@mui/material";
+import { InsertSchema } from "@/supabase";
+import supabase from "@/supabase";
+
+type PersonType = Database["public"]["Enums"]["gender"];
+type FieldType = string | (boolean | null) | booleanGroup | Address | People[];
+type EventType = React.ChangeEvent<HTMLInputElement> | SelectChangeEvent;
+type OnChangeType = (event: EventType) => void;
+type SetErrorType = (errorKey: string, errorMessage: string) => void;
+type SetFieldType = (fieldKey: string, newFieldValue: FieldType) => void;
+export type ClientDatabaseRecord = InsertSchema["clients"];
+type FamilyDatabaseRecord = InsertSchema["families"];
+
+interface Address {
+    line1: string;
+    line2: string;
+    town: string;
+    county: string;
+    postcode: string;
+}
+
+export interface People {
+    personType: PersonType;
+    age?: number | null;
+    quantity?: number;
+}
+
+export interface ErrorMessages {
+    fullName: string;
+    phoneNumber: string;
+    addressLine1: string;
+    addressPostcode: string;
+    adults: string;
+    numberChildren: string;
+    nappyErrorMessage: string;
+}
+
+export interface Fields {
+    fullName: string;
+    phoneNumber: string;
+    addressLine1: string;
+    addressLine2: string;
+    addressTown: string;
+    addressCounty: string;
+    addressPostcode: string;
+    adults: People[];
+    numberChildren: number;
+    children: People[];
+    dietaryRequirements: booleanGroup;
+    feminineProducts: booleanGroup;
+    babyProducts: boolean | null;
+    nappySize: string;
+    petFood: booleanGroup;
+    otherItems: booleanGroup;
+    deliveryInstructions: string;
+    extraInformation: string;
+}
+
+export const setErrorFunction = (
+    errorSetter: React.Dispatch<React.SetStateAction<ErrorMessages>>,
+    errorValues: ErrorMessages
+): SetErrorType => {
+    return (errorKey, errorMessage): void => {
+        errorSetter({ ...errorValues, [errorKey]: errorMessage });
+    };
+};
+
+export const setFieldFunction = (
```

Comment:
these don't need to be suffixed by Function (although it's fine if this is intentional)